packages/next/src/cli/next-dev.ts

```diff
@@ -108,8 +119,11 @@ const handleSessionStop = async (signal: string | null) => {
   process.exit(0)
 }
 
-process.on('SIGINT', () => handleSessionStop('SIGINT'))
-process.on('SIGTERM', () => handleSessionStop('SIGTERM'))
+process.on('SIGINT', () => handleSessionStop('SIGKILL'))
+process.on('SIGTERM', () => handleSessionStop('SIGKILL'))
```

Comment:
a little unsure about switching this to SIGKILL because it could potentially mess with someone using a custom signal but it looks like that was never supported in `next dev` anyway, at least officially:

https://nextjs.org/docs/pages/building-your-application/deploying

> **Good to know**: Manual signal handling is not available in `next dev`.

Also because of the block in `packages/next/src/bin/next.ts` that's removed in this MR, it has the extra check `command !== 'dev'`, so behavior should effectively be the same (exit the child immediately when the dev server receives SIGINT or SIGTERM)